File: 1537-maximum-score-after-splitting-a-string/maximum-score-after-splitting-a-string.cpp
class Solution {
public:
    int maxScore(string s) {
        // 01.01.25 POTD
        int ans=0, n=s.size();
        int oneCount = 0, zeroCount = '1'-s[0];

        for(int i=n-1; i>0; i--){
            oneCount += s[i]-'0';
        }

        for(int i=1; i<n; i++){
            ans = max(ans, zeroCount + oneCount);
            oneCount -= (s[i]-'0');
            zeroCount += ('1'-s[i]);
        }

        return ans;
    }
};

File: 1537-maximum-score-after-splitting-a-string/maximum-score-after-splitting-a-string.py
class Solution:
    def maxScore(self, s: str) -> int:
        # 01.01.25 POTD
        ans = 0
        ones = 0
        zeros = 1 if s[0] == '0' else 0
        n = len(s)

        for i in range(n-1,0,-1):
            ones += int(s[i])

        for i in range(1,n):
            ans = max(ans, zeros + ones)
            zeros += 1 if s[i] == '0' else 0
            ones -= int(s[i])

        return ans